Tests/JetworkingTests/Mocks/MockURLProtocol.swift

```diff
@@ -0,0 +1,49 @@
+import Foundation
+
+/// Inspired by https://medium.com/@dhawaldawar/how-to-mock-urlsession-using-urlprotocol-8b74f389a67a
+final class MockURLProtocol: URLProtocol {
+    enum MockError: Error {
+        case invalidConfiguration
+    }
+
+    static var requestHandler: ((URLRequest) throws -> (HTTPURLResponse, Data?, TimeInterval))?
+
+    override class func canInit(with request: URLRequest) -> Bool {
+       // To check if this protocol can handle the given request.
+       return true
+     }
+
+     override class func canonicalRequest(for request: URLRequest) -> URLRequest {
+       // Here you return the canonical version of the request but most of the time you pass the orignal one.
+       return request
+     }
+
+     override func startLoading() {
+        guard let handler = MockURLProtocol.requestHandler else { return }
+
+        do {
+            // 2. Call handler with received request and capture the tuple of response and data.
```

Comment:
Where is the first point? Do we need it? Is only the comment missing?

Reply:
Removed the enumeration 😉

Reply:
Okay.